docs/pages/database-access/guides/redis.mdx

```diff
@@ -0,0 +1,350 @@
+
+## Prerequisites
+
+- Teleport version `9.0` or newer.
+- Redis database (standalone or cluster) version `6.0` or newer.
+- `redis-cli` installed and added to your system's `PATH` environment variable.
+
+<Admonition type="note" title="Note">
+  Redis `7.0` and RESP3 (REdis Serialization Protocol) are currently not supported.
+</Admonition>
+
+## Step 1/5. Set up Teleport Auth and Proxy
+
+(!docs/pages/includes/database-access/start-auth-proxy.mdx!)
+
+## Step 2/5. Create a Teleport user
+
+(!docs/pages/includes/database-access/create-user.mdx!)
+
+## Step 3/5. Create Redis users.
+
+Redis user must be protected by a strong password. We recommend using a password generator built-in in Redis.
+To generate password connect to a Redis instance and call:
+
+```
+ACL GENPASS
+```
+
+or use OpenSSL if you have it installed on your machine
+
+```shell
+openssl rand -hex 32
+```
+
+This command generates a password like that `0a49c95a25b0f92787d07a5b6bdc1a25b3a2130a1b338ec524f1b5dcd708bc65`.
+
+Create `users.acl` file with users. File defines Redis users, passwords required to log in as a given user
+and sets of Redis ACL rules. Redis allows providing passwords as a plain text or SHA256 checksum version.
+We strongly recommend to use SHA256 checksum instead of plain text password.
+
+<Tabs>
+  <TabItem label="Standalone">
+    ```text
+    user alice on #57639ed88a85996453555f22f5aa4147b4c9614056585d931e5d976f610651e9 allcommands allkeys
+    user default off
+    ```
+
+    For more ACL examples refer to [Redis documentation](https://redis.io/topics/acl).
+  </TabItem>
+  <TabItem label="Cluster">
+    Redis in Cluster mode requires a special user created on master nodes to enable master-replica communication.
+    ```text
+    user alice on #57639ed88a85996453555f22f5aa4147b4c9614056585d931e5d976f610651e9 allcommands allkeys
+    user replica-user on #42a9798b99d4afcec9995e47a1d246b98ebc96be7a732323eee39d924006ee1d &* -@all +replconf +ping +psync
+    user default off
+    ```
+
+    For more ACL examples refer to Redis documentation: [ACL](https://redis.io/topics/acl), [ACL rules for replicas](https://redis.io/topics/acl#acl-rules-for-sentinel-and-replicas).
+  </TabItem>
+</Tabs>
+
+<Admonition type="warning">
+  It's very important to either disable or protect with password the `default` user. Otherwise, everyone with access
+  to the database can login as the `default` user which by default has administrator privileges.
+</Admonition>
+
+## Step 4/5
+
+### Set up mutual TLS
+
+(!docs/pages/includes/database-access/tctl-auth-sign.mdx!)
+
+Create the secrets:
+
+<Tabs>
+  <TabItem label="Standalone">
+    When connecting to standalone Redis, sign the certificate for the hostname over
+    which Teleport will be connecting to it.
+
+    For example, if your Redis server is accessible at `redis.example.com`
+    hostname, run:
+
+    ```code
+    $ tctl auth sign --format=redis --host=redis.example.com --out=server --ttl=2190h
```

Comment:
Any reason we default to 3 months? Am I correct in thinking that after 3 months the Redis integration would no longer work.

Reply:
@benarent It requires you to generate a new cert, but we're using the same value in all our DB integrations, example from Mongo https://goteleport.com/docs/ver/9.0/database-access/guides/mysql-self-hosted/
If we want to change it I think we should create a separate issue and change the ttl for all integrations.

@r0mant @smallinsky Any comments?

Reply:
Not sure why `2190h` was picked but agree that If we want to change that we should do this be a separate PR where values for all DBs will be updated.